torchbenchmark/models/pytorch_unet/metadata.yaml

```diff
@@ -5,4 +5,4 @@ eval_benchmark: false
 eval_deterministic: true
 eval_nograd: true
 train_benchmark: false
-train_deterministic: true
+train_deterministic: false
```

Comment:
Why all the removal you are proposing what it will be the scope of this metadata?
Also who is controlling `eager_two_runs_differ`? See https://github.com/pytorch/pytorch/pull/121324#issuecomment-2004585356

Reply:
@bhack The metadata would be default value of pytorch, which is `False` for both `train` and `eval`.

Reply:
`torchbench.py` is controlling `eager_two_runs_differ`: https://github.com/pytorch/pytorch/blob/76a87e33a0fdd0639842218ce26943e7b4f3838b/benchmarks/dynamo/common.py#L2478

Reply:
Is it on this repo too?
https://github.com/search?q=repo%3Apytorch%2Fbenchmark%20eager_two&type=code

Reply:
> Is it on this repo too? [github.com/search?q=repo%3Apytorch%2Fbenchmark%20eager_two&type=code](https://github.com/search?q=repo%3Apytorch%2Fbenchmark%20eager_two&type=code)

The version in `pytorch/pytorch` is the ground truth, and the pytorch/benchmark version will be auto-sync-ed from `pytorch/pytorch`.

Reply:
But who is going to define if two unet eager runs need to be the same or not?

Reply:
It seems that it could be deterministic when compiled if I have interpreted this correctly:
https://github.com/pytorch/pytorch/pull/121769#issuecomment-2030738866

What do you think?

Reply:
> But who is going to define if two unet eager runs need to be the same or not?

Sorry about the confusion. Yes, it is defined by `pytorch/benchmark`: https://github.com/pytorch/benchmark/blob/main/test.py#L72.  If two eager runs differ, the `pytorch/benchmark` CI will fail.

Reply:
So as it seems to be deterministic only compiled, at least in training, what do you want to do here and in the pytorch repo?

Reply:
In our previous tests it is deterministic in the eager mode. If there is a PR changes this behavior and makes it only deterministic when compiled and non-deterministic in eager mode, it is up to the PyTorch team to decide if they should accept it. If the PyTorch Dev Infra team accepts it, we can skip the eager-mode deterministic test for this model.